app/Services/EmployeeService.php

```diff
@@ -18,31 +19,61 @@ public function getAll()
         return $this->employeeRepositoryInterface->getAll();
     }
 
-    public function find($id)
+    public function find($uuid)
     {
-        return $this->employeeRepositoryInterface->find($id);
+        return $this->employeeRepositoryInterface->find($uuid);
+    }
+
+    public function findSlug($name)
+    {
+        $slug = Str::slug($name,'_');
+        return $this->employeeRepositoryInterface->findSlug($slug);
     }
 
     public function store($request)
     {
-        $request['tgl_lahir'] = date_format(date_create_from_format('d/m/Y', $request['tgl_lahir']), 'Y-m-d');
+        $request['slug'] = Str::slug($request['nama'], '_')
+        . (($count = count($this->findSlug($request['nama']))) > 0 ? 
+            '_' . $count+1 : '') ;
+        $request['uuid'] = Uuid::uuid4()->getHex();
```

Comment:
$request['tgl_lahir'] = Carbon::createFromFormat('d/m/Y', $request['tgl_lahir'])->format('Y-m-d');
```suggestion
        $request['tgl_lahir'] = date_create_from_format('d/m/Y', $request['tgl_lahir'])->format('Y-m-d');
```